### PR TITLE
    Use ytt to create pipelines

### PR DESCRIPTION
**Pull request by beeender**
- Rewrite the pipelines by using ytt for code reuse.
- Add `fly.sh` for easier manipulating with pipelines.
- Create configs for pr, commit and dev pipelines.
- Fix the test task for rhel8.
- Use build/test images pair for all distros. And modify the
  `build_diskquota.sh` so it won't need `Python.h` during the configure
  stage.
- Fix the pipiline bug which a `ubuntu` test may use a `rhel6` diskquota
  binary to do the test. That is caused by the same name of the task
  output.

To replace #115

### CODE REVIEW

**Comment by beeender** [comment]
A testing dev pipeline can be seen at https://extensions.ci.gpdb.pivotal.io/teams/main/pipelines/DEV:mulong_diskquota/jobs/build_test/builds/8

**Review by xuebinsu — APPROVED** [review]
LGTM

File: concourse/scripts/build_diskquota.sh [hunk 1]
```diff
@@ -20,43 +20,30 @@ function pkg() {
...
     chmod a+x install_gpdb_component
+    install_files=( \
+        "lib/postgresql/diskquota.so" \
+        "share/postgresql/extension/diskquota.control" \
+        "share/postgresql/extension/diskquota--1.0.sql" \
+        "share/postgresql/extension/diskquota--2.0.sql" \
+        "share/postgresql/extension/diskquota--1.0--2.0.sql" \
+        "share/postgresql/extension/diskquota--2.0--1.0.sql" \
+        "install_gpdb_component")
     case "$DISKQUOTA_OS" in
```
Comment (higuoxing) [inline concourse/scripts/build_diskquota.sh:31]: Looks that the variable `DISKQUOTA_OS` is set and immediately used in [concourse/pipeline/job_def.lib.yml](https://github.com/greenplum-db/diskquota/pull/143/files#diff-c09e583fd3d13aeb15ba4b34f59fa998910e4717884d1d8633277dfbc75a6bf9). Can we simplify this `case` statement with the following script?

```
tar -czf $TOP_DIR/diskquota_artifacts/diskquota-${DISKQUOTA_VERSION}-${DISKQUOTA_OS}_x86_64.tar.gz \
    "${install_files[@]}"
```

**Review by higuoxing — COMMENTED** [review]
LGTM with one nit.

File: concourse/scripts/build_diskquota.sh (hunk 1 above)
Comment (beeender) [reply concourse/scripts/build_diskquota.sh:31]: good idea, will change it

**Review by liuxueyang — APPROVED** [review]

File: concourse/scripts/build_diskquota.sh (hunk 1 above)
Comment (beeender) [reply concourse/scripts/build_diskquota.sh:31]: I will create a separate PR for that

**Pull request merged**